File: app/src/main/java/com/fed/notes/di/AppComponent.java
package com.fed.notes.di;

import com.fed.notes.view.ListFragment;
import com.fed.notes.view.NoteEditorFragment;
import com.fed.notes.view.NotePreviewFragment;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by Fedor SURIN on 03.11.2017.
 */

@Singleton
@Component(modules = AppModule.class)
public interface AppComponent {

    void inject(NotePreviewFragment notePreviewFragment);

    void inject(NoteEditorFragment noteEditorFragment);

    void inject(ListFragment listFragment);
}

File: app/src/main/java/com/fed/notes/di/AppComponent.kt
package com.fed.notes.di

import com.fed.notes.presenter.EditorPresenter
import com.fed.notes.presenter.PreviewPresenter
import com.fed.notes.view.ListFragment
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class))
interface AppComponent {

    fun inject(listFragment: ListFragment)

    fun inject(previewPresenter: PreviewPresenter)

    fun inject(editorPresenter: EditorPresenter)
}
